src/compound/SupplyVaultBase.sol

```diff
@@ -1,139 +1,189 @@
 import {ERC20, SafeTransferLib} from "@rari-capital/solmate/src/utils/SafeTransferLib.sol";
-import {CompoundMath} from "@morpho-labs/morpho-utils/math/CompoundMath.sol";
 import {Types} from "@contracts/compound/libraries/Types.sol";
 
-import {ERC4626UpgradeableSafe, ERC20Upgradeable} from "../ERC4626UpgradeableSafe.sol";
+import {ERC4626UpgradeableSafe, ERC4626Upgradeable, ERC20Upgradeable} from "../ERC4626UpgradeableSafe.sol";
 
 /// @title SupplyVaultBase.
 /// @author Morpho Labs.
 /// @custom:contact security@morpho.xyz
 /// @notice ERC4626-upgradeable Tokenized Vault abstract implementation for Morpho-Compound.
-abstract contract SupplyVaultBase is ERC4626UpgradeableSafe, OwnableUpgradeable {
-    using CompoundMath for uint256;
+abstract contract SupplyVaultBase is ISupplyVaultBase, ERC4626UpgradeableSafe, OwnableUpgradeable {
     using SafeTransferLib for ERC20;
 
+    /// EVENTS ///
+
+    /// @notice Emitted when MORPHO rewards are transferred to `recipient`.
+    /// @param recipient The recipient of the rewards.
+    /// @param amount The amount of rewards transferred.
+    event RewardsTransferred(address recipient, uint256 amount);
+
     /// ERRORS ///
 
-    /// @notice Thrown when the zero address is passed as input.
+    /// @notice Thrown when the zero address is passed as input or is the recipient address when calling `transferRewards`.
     error ZeroAddress();
 
-    /// STORAGE ///
+    /// IMMUTABLES ///
 
     IMorpho public immutable morpho; // The main Morpho contract.
     address public immutable wEth; // The address of WETH token.
     ERC20 public immutable comp; // The COMP token.
+    ERC20 public immutable morphoToken; // The address of the Morpho Token.
+    ILens public immutable lens; // The address of the Morpho Lens.
+    address public immutable recipient; // The recipient of the rewards that will redistribute them to vault's users.
+
+    /// STORAGE ///
 
     address public poolToken; // The pool token corresponding to the market to supply to through this vault.
 
     /// CONSTRUCTOR ///
 
     /// @dev Initializes network-wide immutables.
     /// @param _morpho The address of the main Morpho contract.
-    constructor(address _morpho) {
+    /// @param _morphoToken The address of the Morpho Token.
+    /// @param _lens The address of the Morpho Lens.
+    /// @param _recipient The recipient of the rewards that will redistribute them to vault's users.
+    constructor(
+        address _morpho,
+        address _morphoToken,
+        address _lens,
+        address _recipient
+    ) {
+        if (
+            _morpho == address(0) ||
+            _morphoToken == address(0) ||
+            _lens == address(0) ||
+            _recipient == address(0)
+        ) revert ZeroAddress();
         morpho = IMorpho(_morpho);
-        wEth = morpho.wEth();
+        wEth = morpho.wEth(); // Reverts if morpho is zero address, so no zero address check is needed.
```

Comment:
but we still check it ^^
I would remove the comment, as the gas saving it pointless here

Reply:
Ok, I did this: #242 
But feel free to close it

Reply:
Not changing the bytecode so I'm ok with it